docs/Theory_cluster_batch_scheduling.rst

```diff
@@ -1,15 +1,132 @@
+that can be scheduled independently on the cluster but can run in
+parallel if there are resources available. The distinction between
+steps and tasks is that multiple tasks, if they are part of the same
+step, cannot depend on results of other task because there are no
+guarantees on the order in which they will be executed.
+
+Finally each process group is the basic unit that is scheduled in the
+cluster. It comprises of a set of processes (or threads) that can run
+on a number of resources (CPU, GPU, RAM, ...) and are scheduled
+together as a unit on one or more machines.
+
+Each of these concepts lends itself to a particular use. For multi-gpu
+training in AI workloads you would use one task per GPU for data
+paralellism or one process group if you are doing model
+parallelism. Hyperparameter optimisation can be done using a
+combination of tasks and steps but is probably better left to a
+framework outside of the scope of the workload manager.
+
+If this all seems complicated, you should know that all these things
+do not need to always be used. It is perfectly acceptable to sumbit
+jobs with a single step, a single task and a single process.
+
+
+Understanding the queue
+-----------------------
+
+The available resources on the cluster are not infinite and it is the
+workload manager's job to allocate them. Whenever a job request comes
+in and there are not enough resources available to start it
+immediately, it will go in the queue.
+
+Once a job is in the queue, it will stay there until another job
+finishes and then the workload manager will try to use the newly freed
+resources with jobs from the queue. The exact order in which the jobs
+will start is not fixed, because it depends on the local policies
+which can take into account the user priority, the time since the job
+was requested, the amount of resources requested and possibly other
+things. There should be a tool that comes with the manager where you
+can see the status of your queued jobs and why they remain in the
+queue.
+
+
+About partitions
+----------------
+
+The workload manger will divide the cluster into partitions according
+to the configuration set by the admins. A partition is a set of
+machines typically reserved for a particular purpose. An example might
+be CPU-only machines for preprocessing setup as a separate partition.
+It is possible for multiple partitions to share resources.
+
+There will always be at least one partition that is the default
+partition in which jobs without a specific request will go. Other
+partitions can be requested, but might be restricted to a group of
+users, depending on policy.
+
+Partitions are useful for a policy standpoint to ensure efficient use
+of the cluster resources and avoid using up too much of one resource
+type blocking use of another. They are also useful for heterogenous
+clusters where different hardware is mixed in and not all software is
+compatible with all of it (for example x86 and POWER cpus).
```

Comment:
While I totally agree, note that this is not our use of partitions yet. Currently all GPU types, x86 and Power9 are mixed in the same partitions.
However this is the path I want to take for the Mila cluster.